### PR TITLE
.Net: Use GetType and typeof for loggers

### PR DESCRIPTION
**Pull request by lemillermicrosoft**
### Motivation and Context
This pull request updates the logger creation in various classes throughout the project, replacing the usage of `nameof` with `typeof` to ensure that the logger instances are created with the correct type information.

When objects create loggers using only the class or type name results in the need to control filtering for each individual class or type. If we instead use the full name of the class or type, we can depend on namespace level filtering.

```
builder.AddFilter("Microsoft", LogLevel.Warning);
builder.AddFilter("Microsoft.SemanticKernel", LogLevel.Critical);
builder.AddFilter("Microsoft.SemanticKernel.Reliability", LogLevel.Information);
```

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/samples/KernelSyntaxExamples/Program.cs [hunk 1]
```diff
@@ -20,61 +20,61 @@ public static async Task Main()
...
+        // await Example46_Weaviate.RunAsync().SafeWaitAsync(cancelToken);
+        // await Example47_Redis.RunAsync().SafeWaitAsync(cancelToken);
+        // await Example48_GroundednessChecks.RunAsync().SafeWaitAsync(cancelToken);
+        // await Example49_LogitBias.RunAsync().SafeWaitAsync(cancelToken);
+        // await Example50_Chroma.RunAsync().SafeWaitAsync(cancelToken);
+        // await Example51_StepwisePlanner.RunAsync().SafeWaitAsync(cancelToken);
+        // await Example52_ApimAuth.RunAsync().SafeWaitAsync(cancelToken);
+        // await Example53_Kusto.RunAsync().SafeWaitAsync(cancelToken);
+        // await Example54_AzureChatCompletionWithData.RunAsync().SafeWaitAsync(cancelToken);
+        // await Example55_TextChunker.RunAsync().SafeWaitAsync(cancelToken);
```
Comment (stephentoub) [inline dotnet/samples/KernelSyntaxExamples/Program.cs:77]: Did you mean to include this commenting?

Comment (lemillermicrosoft) [reply dotnet/samples/KernelSyntaxExamples/Program.cs]: Nope, resolved.

**Comment by dmytrostruk** [comment]
@lemillermicrosoft I like this proposal, I used the same approach for metering and tracing, so namespace level filtering is available there as well. This provides granular control over telemetry for different functionality.

Maybe we can also add some information about namespace level filtering for logging in document about telemetry in scope of this PR:
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/docs/TELEMETRY.md

**Review by dmytrostruk — APPROVED** [review]

**Comment by dmytrostruk** [comment]
@lemillermicrosoft I also tested this change manually with Application Insights and it works as expected:
image

Now `CategoryName` field contains full name of specific SK class/module, which should improve observability experience a lot.
Thanks again!

**Review by SergeyMenshykh — APPROVED** [review]

**Pull request merged**